job/job.go

```diff
@@ -232,11 +232,30 @@ func (sg *scopedGroup) Add(jobs ...Job) {
 	sg.group.add(sg.health, jobs...)
 }
 
-var nameRegex = regexp.MustCompile(`^[a-zA-Z][a-zA-Z0-9_\-]{0,100}$`)
-
-func validateName(name string) error {
-	if !nameRegex.MatchString(name) {
-		return fmt.Errorf("invalid job name: %q, expected to match %q", name, nameRegex)
+const maxNameLength = 100
+
+func sanitizeName(name string) string {
+	runes := []rune(name)
+	mangled := false
+	for i, r := range runes[:min(maxNameLength, len(name))] {
+		switch {
+		case r >= 'a' && r <= 'z':
+		case r >= 'A' && r <= 'Z':
+		case r >= '0' && r <= '9':
+		case r == '-' || r == '_':
+		default:
+			// Replace unallowed characters with underscores.
+			mangled = true
+			runes[i] = '_'
+		}
 	}
-	return nil
+	if mangled || len(runes) > maxNameLength {
+		// Name was mangled or is too long, truncate and append hash.
+		const hashLen = 10
+		hash := fmt.Sprintf("%x", sha256.Sum256([]byte(name)))
```

Comment:
Might want to avoid some "bad" words here, c. f. the cilium/cilium fix in https://github.com/cilium/cilium/pull/35031#event-14397152235

Reply:
What are you referring to? What's a bad word?